ccl/changefeedccl: TestChangefeedProtectedTimestampUpdate failed

ccl/changefeedccl.TestChangefeedProtectedTimestampUpdate [failed](https://tanzanite.cluster.engflow.com/invocations/default/67b4a299-1729-4791-be66-88b87b330809?testReportRun=2&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [13d8c5191614530b50802f77081f15d2ce2ca705](https://github.com/cockroachdb/cockroach/commits/13d8c5191614530b50802f77081f15d2ce2ca705):


```
=== RUN   TestChangefeedProtectedTimestampUpdate
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedProtectedTimestampUpdate1283828548
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:10100: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedProtectedTimestampUpdate1283828548
--- FAIL: TestChangefeedProtectedTimestampUpdate (32.94s)
=== RUN   TestChangefeedProtectedTimestampUpdate/kafka
    helpers_test.go:914: making server as system tenant
    helpers_test.go:990: making kafka feed factory
    changefeed_test.go:10088: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:10088
        	            				pkg/ccl/changefeedccl/helpers_test.go:1162
        	            				pkg/ccl/changefeedccl/helpers_test.go:1197
        	Error:      	"1738937654232989092.0000000000" is not less than "1738937654232989092.0000000000"
        	Test:       	TestChangefeedProtectedTimestampUpdate/kafka
    --- FAIL: TestChangefeedProtectedTimestampUpdate/kafka (32.93s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedProtectedTimestampUpdate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47295

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -10081,6 +10081,9 @@ func TestChangefeedProtectedTimestampUpdate(t *testing.T) {
 		changefeedbase.ProtectTimestampLag.Override(
 			context.Background(), &s.Server.ClusterSettings().SV, 10*time.Millisecond)
 
+		// Ensure that the resolved timestamp advances at least once
+		// since the PTS lag override.
+		testutils.SucceedsSoon(t, checkHWM)
 		testutils.SucceedsSoon(t, checkHWM)
 
 		sqlDB.QueryRow(t, ptsQry).Scan(&ts2)

